src/QMCWaveFunctions/Fermion/DiracDeterminant.cpp

```diff
@@ -43,7 +44,14 @@ DiracDeterminant<DU_TYPE>::DiracDeterminant(std::unique_ptr<SPOSet>&& spos,
   resize(NumPtcls, NumPtcls);
 
   if (isOptimizable())
-    Phi->buildOptVariables(NumPtcls);
+  {
```

Comment:
We can remove `if (isOptimizable())`
Also don't throw `throw std::runtime_error("Cast of Phi to RotatedSPOs failed.");`
If the SPOSet is optimizable but not via RotatedSPOs. The code should not stop.